Framework/DataHandling/src/SaveAscii2.cpp

```diff
@@ -443,11 +443,11 @@ void SaveAscii2::populateSpectrumNumberMetaData() {
 void SaveAscii2::populateAngleMetaData() {
   std::vector<std::string> angles;
   const size_t nHist = m_ws->getNumberHistograms();
+  const auto &spectrumInfo = m_ws->spectrumInfo();
   for (size_t i = 0; i < nHist; i++) {
     const auto specNo = m_ws->getSpectrum(i).getSpectrumNo();
     const auto workspaceIndex = m_specToIndexMap[specNo];
```

Comment:
I don't know, I think we are just doing what you say but I am getting confused. I've made some changes, let me know if you think they are OK.